ext/opentelemetry-ext-stackdriver/src/opentelemetry/ext/stackdriver/trace/__init__.py

```diff
@@ -0,0 +1,281 @@
+            logger.warning("Error while writing to stackdriver: %s", ex)
+            return SpanExportResult.FAILED_RETRYABLE
+
+        return SpanExportResult.SUCCESS
+
+    def translate_to_stackdriver(
+        self, spans: typing.Sequence[Span]
+    ) -> typing.List[typing.Dict[str, typing.Any]]:
+        """Translate the spans to Stackdriver format.
+
+        Args:
+            spans: Tuple of spans to convert
+        """
+
+        stackdriver_spans = []
+
+        for span in spans:
+            ctx = span.get_context()
+            trace_id = "{:032x}".format(ctx.trace_id)
+            span_id = "{:016x}".format(ctx.span_id)
+            span_name = "projects/{}/traces/{}/spans/{}".format(
+                self.project_id, trace_id, span_id
+            )
+
+            parent_id = None
+            if isinstance(span.parent, trace_api.Span):
+                parent_id = "{:016x}".format(span.parent.get_context().span_id)
+            elif isinstance(span.parent, trace_api.SpanContext):
+                parent_id = "{:016x}".format(span.parent.span_id)
+
+            start_time = None
+            if span.start_time:
+                start_time = ns_to_iso_str(span.start_time)
+            end_time = None
+            if span.end_time:
+                end_time = ns_to_iso_str(span.end_time)
+
+            span.attributes["g.co/agent"] = AGENT
+            attr_map = extract_attributes(span.attributes)
+
+            sd_span = {
+                "name": span_name,
+                "spanId": span_id,
+                "parentSpanId": parent_id,
+                "displayName": get_truncatable_str(span.name),
+                "attributes": map_attributes(attr_map),
+                "links": extract_links(span.links),
+                "status": extract_status(span.status),
+                "timeEvents": extract_events(span.events),
+                "startTime": start_time,
+                "endTime": end_time,
+            }
+
+            stackdriver_spans.append(sd_span)
+
+        return stackdriver_spans
+
+    def shutdown(self):
+        pass
+
+
+def get_truncatable_str(str_to_convert):
+    """Truncate a string if exceed limit and record the truncated bytes
+        count.
+    """
+    truncated, truncated_byte_count = check_str_length(
+        str_to_convert, MAX_LENGTH
+    )
+
+    result = {
+        "value": truncated,
+        "truncated_byte_count": truncated_byte_count,
+    }
+    return result
+
+
+def check_str_length(str_to_check, limit=MAX_LENGTH):
+    """Check the length of a string. If exceeds limit, then truncate it.
+    """
+    str_bytes = str_to_check.encode("utf-8")
+    str_len = len(str_bytes)
+    truncated_byte_count = 0
+
+    if str_len > limit:
+        truncated_byte_count = str_len - limit
+        str_bytes = str_bytes[:limit]
+
+    result = str(str_bytes.decode("utf-8", errors="ignore"))
+
+    return (result, truncated_byte_count)
+
+
+def extract_status(status: trace_api.Status):
```

Comment:
Do you think it's necessary to remove all the single-use functions? Speaking because Jaeger and Zipkin exporters have some of them too (Example in Jaeger:  `_extract_refs_from_span` `_extract_logs_from_span`)... And maybe they need a second look.

Reply:
I don't think it is necessary, but it is very convenient :slightly_smiling_face: 

They may be present somewhere else, but that does not mean we should have them here, of course. Removing them means less code, less stuff to document, but above all, it means that when you are reading code you don't have to stop, read code somewhere else (maybe doing some mental replacement of variable names that were passed as arguments) and then come back to the original point in the code.

Reply:
Makes sense. Thank you for taking the time to explain!